src/test/python/deep_qa_test/layers/tuple_match_test.py

```diff
@@ -0,0 +1,76 @@
+# pylint: disable=no-self-use
+from unittest import TestCase
+import numpy as np
+from numpy.testing import assert_array_almost_equal
+from scipy.stats import logistic
+
+from keras import backend as K
+from keras.layers import Input
+from keras.models import Model
+from deep_qa.layers.entailment_models.tuple_match import TupleMatch
+
+class TestTupleMatch(TestCase):
+
+    def setUp(self):
+        num_slots = 3
+        t1_words_per_slot = 4
+        t2_words_per_slot = 3
+        self.tuple1_input = Input(shape=(num_slots, t1_words_per_slot), dtype='float32', name="input_tuple1")
+        self.tuple2_input = Input(shape=(num_slots, t2_words_per_slot), dtype='float32', name="input_tuple2")
+        self.tuple_match_params = {'num_hidden_layers': 1, 'hidden_layer_width': 1,
+                                   'hidden_layer_activation': 'linear', "num_entailment_features": 3}
+        self.tuple1 = np.array([[[1, 2, 0, 0],
+                                 [1, 2, 3, 0],
+                                 [0, 0, 0, 0]]])
+        self.tuple2 = np.array([[[2, 1, 1],
+                                 [1, 2, 0],
+                                 [4, 5, 0]]])
+
+
+    def test_general_case(self):
+
+        match_layer = TupleMatch(self.tuple_match_params)
+        output = match_layer([self.tuple1_input, self.tuple2_input])
+        model = Model([self.tuple1_input, self.tuple2_input], output)
+
+        # Get the initial weights for use in testing
+        dense1_hidden_weights = K.eval(model.trainable_weights[0])
+        dense1_biases = K.eval(model.trainable_weights[1])
+        score_hidden_weights = K.eval(model.trainable_weights[2])
+        score_bias = K.eval(model.trainable_weights[3])
+
+        # Testing general unmasked case.
+        desired_overlap = np.array([[3/2,
+                                     2/3,
+                                     0]])
+        # Desired_overlap gets fed into the inner NN.
+        dense1_activation = np.dot(desired_overlap, dense1_hidden_weights) + dense1_biases
+        final_score = np.dot(dense1_activation, score_hidden_weights) + score_bias
+        desired_result = logistic.cdf(final_score)
+        result = model.predict([self.tuple1, self.tuple2])
+        assert_array_almost_equal(result, desired_result)
+
+    def test_masks_handled_correctly(self):
+        # tuple1 has shape (b, num_slots, t1_words_per_slot), so mask1 has shape (b, num_slots)
+        mask1 = K.variable(np.array([1, 1, 0]))
```

Comment:
You want this to take tuple inputs, with the mask being `None`, instead of specifying a mask.